cloudbridge/cloud/providers/gce/services.py

```diff
@@ -290,48 +312,54 @@ def get(self, image_name):
             # The API will throw an TypeError, if parameter `image` does not
             # match the pattern "[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?".
             print("TypeError: {0}".format(type_error))
-            return None
         except googleapiclient.errors.HttpError as http_error:
             print("googleapiclient.errors.HttpError: {0}".format(http_error))
-            return None
+        for public_image in self._public_images:
+            if public_image.name == image_name:
+                return public_image
+        return None
 
     def find(self, name, limit=None, marker=None):
         """
         Searches for an image by a given list of attributes
         """
+        if self._public_images == None:
+            self._retrieve_public_images()
         filters = {'name': name}
+        images = []
         try:
             response = self.provider.gce_compute \
                                     .images() \
                                   .list(project=self.provider.project_name) \
                                   .execute()
+            if 'items' in response:
+                images = [GCEMachineImage(self.provider, image) for image
+                          in response['items']]
         except googleapiclient.errors.HttpError as http_error:
             print("googleapiclient.errors.HttpError: {0}".format(http_error))
-            return []
-        if 'items' not in response:
-            return []
-        images = [GCEMachineImage(self.provider, image) for image
-                  in response['items']
-                  if 'name' in image and image['name'] == filters['name']]
+        images.extend(self._public_images)
```

Comment:
This always adds the public images to the response, even if a name filter is provided?
If that is the correct logic, please add a comment here explaining it because it is not obviously correct.